static/app/components/events/interfaces/request/getTransformedData.tsx

```diff
@@ -1,42 +1,35 @@
-import {defined} from 'sentry/utils';
+import {
+  getChildMetaContainer,
+  MetaContainer,
+} from 'sentry/components/events/meta/metaContainer';
+import {EntryRequest} from 'sentry/types';
 
-function getTransformedData(data: any, meta: Record<any, any> | undefined) {
-  if (Array.isArray(data)) {
-    return data
-      .filter(dataValue => {
-        if (typeof dataValue === 'string') {
-          return !!dataValue;
-        }
-
-        return defined(dataValue);
-      })
-      .map((dataValue, index) => {
-        if (Array.isArray(dataValue)) {
-          return {
-            data: dataValue,
-            meta: meta?.[index]?.[1]?.[1]?.[''],
-          };
-        }
-
-        if (typeof data === 'object') {
-          return {
-            data: Object.keys(dataValue).flatMap(key => [key, dataValue[key]]),
-          };
-        }
```

Comment:
This seems like a buggy case... `typeof data === 'object'` will always be true since we established on L4 that `Array.isArray(data)`... I _think_ this was supposed t be `typeof dataValue`... but even if that was fixed, flatMap is going to be produce a flattened list, so `{a: 1, b: 2}` would turn into `['a', 1, 'b', 2]` which doesn't seem to match up to the expected return type. I could see that the correct behavior would be to produce `[{data: ['a', 1]}, {data: ['b', 2]}]` but this doesn't seem to even be supported case according to the type of `EntryRequest['data']['data']`... so i just deleted it 😬 . 

Did I go to crazy with refactoring this module?